Soundprints/Soundprints/Source/User Interface/Main Map/Views/RecordButtonView.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  RecordButtonView.swift
+//  Soundprints
+//
+//  Created by Peter Korosec on 15/12/2017.
+//  Copyright © 2017 Kamino. All rights reserved.
+//
+
+import UIKit
+
+protocol RecordButtonViewDelegate: class {
+    func recordButtonViewChangedState(sender: RecordButtonView, recording: Bool)
+}
+
+class RecordButtonView: UIView {
+    
+    // TODO: in progress...
+
+    // MARK: - Properties
+    
+    weak var delegate: RecordButtonViewDelegate?
+    
+    private(set) var recording: Bool = false
+    
+    var recordingColor: UIColor = .red
+    var defaultColor: UIColor = .blue
+    var trailColor: UIColor = .white
+    
+    var animationDuration: TimeInterval = 1.2
+    
+    private var trailLifeSpan: TimeInterval {
+        return animationDuration/3
+    }
+    
+    private var boundsCenter: CGPoint {
+        return CGPoint(x: bounds.width/2, y: bounds.height/2)
+    }
+    
+    private var maxRadius: CGFloat {
+        return min(bounds.width, bounds.height)/2.0 - 1.0
+    }
+    
+    var startAngle: CGFloat = 0.0
+    
+    var endAngle: CGFloat {
+        return startAngle + CGFloat.pi*2
+    }
+    
+    // MARK: - Initialization
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+    
+    private func setup() {
+        backgroundColor = .clear
+        
+        let recognizer = UITapGestureRecognizer(target: self, action: #selector(onTap))
+        self.addGestureRecognizer(recognizer)
+    }
+    
+    // MARK: - Lifecycle
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        setNeedsDisplay()
+    }
+    
+    // MARK: - Circle
+    
+    private var circleAnimator: ValueAnimator?
+    
+    // TODO:
```

Comment:
Add some text to this TODO, or remove it if it is not relevant, since it is not clear what is to be done.